src/path_util.rs

```diff
@@ -40,3 +40,36 @@ impl Expand for PathBuf {
         }
     }
 }
+
+pub struct PathsIterator<'a> {
+    path: Option<&'a Path>,
+}
+
+impl<'a> Iterator for PathsIterator<'a> {
+    type Item = &'a Path;
+    fn next(&mut self) -> Option<&'a Path> {
+        match self.path {
+            Some(path) => {
+                if path == Path::new("") {
+                    None
+                } else {
+                    self.path = path.parent();
```

Comment:
I would make `path.parent().unwrap_or(Path::new(""))`, and get rid of `Option` in iterator. We treat `None` and `Path::new("")` the same anyway.